File: csharp/series/Series.cs
using System;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (numbers.Length == 0 || sliceLength > numbers.Length || sliceLength < 1)
            throw new ArgumentException();

        string[] output = new string[numbers.Length - (sliceLength - 1)];
        for (int i = 0; i < numbers.Length - (sliceLength - 1); i++)
        {
            output[i] = numbers.Substring(i, sliceLength);
        }

        return output;
    }
}

File: javascript/series/series.js
//
// This is only a SKELETON file for the 'Series' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Series {
  constructor(numbers) {
    this._digits = [];
    for (let i = 0; i < numbers.length; i++) {
      this._digits.push(parseInt(numbers[i]));
    }
  }

  get digits() {
    return this._digits;
  }

  slices(number) {
    if (number > this._digits.length) {
      throw new Error('Slice size is too big.');
    }

    const output = [];
    for (let i = 0; i < this._digits.length - (number - 1); i++) {
        output.push(this._digits.slice(i, i + number));
    }

    return output;
  }
}
